software/main.py

```diff
@@ -0,0 +1,110 @@
+import image_processor
+import camera
+import motion
+import cv2
+import time
+import numpy as np
+from enum import Enum
+
+class State(Enum):
+    SEARCH_BALL = 1
+    DRIVE_TO_BALL = 2
+    BALL_CENTERED = 3
+
+def main_loop():
+    debug = False# if set to false wont show camera
+    ball_to_right = False
+    
+    #camera instance for normal web cameras
+    #cam = camera.OpenCVCamera(id = 2)
+    # camera instance for realsense cameras
+    cam = camera.RealsenseCamera(exposure = 100, depth_enabled=False)
+    
+    processor = image_processor.ImageProcessor(cam, debug=debug)
+
+    processor.start()
+
+    robot = motion.OmniMotionRobot()
+
+    start = time.time()
+    fps = 0
+    frame = 0
+    frame_cnt = 0
+    cam_width_Right, cam_width_Left = cam.rgb_width / 2 + 20, cam.rgb_width / 2 - 20
```

Comment:
Seems unneeded. Just calculate the middle of the frame and define a offset threshold. to used in below logic.